provider/datagen/src/transform/cldr/currency/mod.rs

```diff
@@ -146,6 +146,7 @@ fn extract_currency_essentials<'data>(
                         short_place_holder_index = USE_ISO_CODE;
                     } else {
                         short_place_holder_index = place_holders.len() as u16;
+                        assert!(short_place_holder_index <= MAX_PLACE_HOLDER_INDEX);
```

Comment:
Suggestion: this function returns a `DataError` so you should return a `DataError` here instead of panicking. Also below.

Reply:
I have filed and todo issue #3900